scripts/data_prep/convert_text_to_mds.py

```diff
@@ -0,0 +1,329 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import math
+import os
+import tempfile
+from argparse import ArgumentParser, Namespace
+from glob import glob
+from multiprocessing import Pool
+from typing import Iterable, List, cast
+
+from composer.utils import (ObjectStore, maybe_create_object_store_from_uri,
+                            parse_uri)
+from streaming import MDSWriter
+from tqdm import tqdm
+from transformers import AutoTokenizer
+
+from llmfoundry.data import ConcatTokensDataset
+from scripts.data_prep.utils import (DownloadingIterable, build_dataloader,
+                                     generate_samples, merge_shard_groups)
+
+
+def parse_args() -> Namespace:
+    """Parse commandline arguments."""
+    parser = ArgumentParser(
+        description=
+        'Convert text files into MDS format, optionally concatenating and tokenizing'
+    )
+    parser.add_argument(
+        '--max_workers',
+        type=int,
+        default=64,
+        help='The maximum number of workers to use for MDS writing')
+    parser.add_argument('--output_folder',
+                        type=str,
+                        required=True,
+                        help='The folder to write output to')
+    parser.add_argument('--input_folder',
+                        type=str,
+                        required=True,
+                        help='The folder with text files to convert to mds')
+    parser.add_argument('--compression',
+                        type=str,
+                        default='zstd',
+                        help='The compression algorithm to use for MDS writing')
+
+    group = parser.add_mutually_exclusive_group(required=False)
+    group.add_argument(
+        '--concat_tokens',
+        type=int,
+        help='Convert text to tokens and concatenate up to this many tokens')
+
+    parser.add_argument('--tokenizer',
+                        type=str,
+                        help='The name of the tokenizer to use')
+    parser.add_argument(
+        '--bos_text',
+        type=str,
+        required=False,
+        default=None,
+        help=
+        'The text to prepend to each example to separate concatenated examples')
+    parser.add_argument(
+        '--eos_text',
+        type=str,
+        required=False,
+        default=None,
+        help=
+        'The text to append to each example to separate concatenated examples')
+    parser.add_argument(
+        '--no_wrap',
+        default=False,
+        action='store_true',
+        help=
+        'Whether to let text examples wrap across multiple training examples')
+    parser.add_argument(
+        '--processes',
```

Comment:
maybe change the name of this to differentiate more clearly from the `max_workers` arg?

Reply:
how about changing max_workers to max_mds_writer_workers?

Reply:
sgtm